File: crates/swiftlet/examples/gr_regular_expression.rs
use std::sync::Arc;
use swiftlet::{ParserConfig, Swiftlet};

fn main() {
    let grammar = r#"
    start: number "+" number
    number: /\d+/
    "#;

    let text = "123+456";

    let conf = Arc::new(ParserConfig::default());
    let parser = Swiftlet::from_str(grammar)
        .map(|grammar| grammar.parser(conf))
        .expect("failed to build parser");
    match parser.parse(&text) {
        Ok(ast) => {
            ast.print()
            // Output: Tree("start", [Tree("number", ["123"]), "+", Tree("number", ["456"])])
        }
        Err(err) => eprintln!("{}", err),
    }
}

File: pyswiftlet/examples/gr_regular_expression.py
"""Grammar regular expression example using the Swiftlet Python bindings."""

from swiftlet import Swiftlet


GRAMMAR = r"""
start: number "+" number
number: /\d+/
"""


def main() -> None:
    parser = Swiftlet(GRAMMAR)
    tree = parser.parse("123+456")
    tree.pretty_print()


if __name__ == "__main__":
    main()
